R/parallel.R

```diff
@@ -0,0 +1,65 @@
+##' Set parallel options.  Having requested more than one core using
+##' [hipercow_resources], here we specify the way that those cores
+##' might be used in parallel.
+##'
+##' @title Specify parallel use of cores
+##'
+##' @param method The parallel method that hipercow will prepare.
+##' Four options are available: `future` will allocate your
+##' reserved cores to a [furrr](https://furrr.futureverse.org/)
+##' cluster; `parallel` will make you a cluster using the
+##' [parallel](https://www.rdocumentation.org/packages/parallel/)
+##' package; `doParallel` will initialise using the
+##' [doParallel](https://cran.r-project.org/web/packages/doParallel/index.html)
+##' package, and finally `NULL`, the default, will do nothing,
+##' and leave you to configure any parallelism yourself. The packages you
+##' use should be provisioned in the usual way with [hipercow_provision].
+##'
+##'
+##' @return A list containing your parallel configuration.
+##'
+##' @export
+hipercow_parallel <- function(method = NULL) {
```

Comment:
so the other half of the problem with parallelism is making sure that the internal cluster has library paths set, packages loaded, and files sourced.  This is very hard to get right without it duplicating a lot of information.  I've been working with Lydia today and have a good handle on what we need now, but that will be additional arguments to add soon.